core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraXCameraUseCase.kt

```diff
@@ -357,6 +365,41 @@ constructor(
                         }
                     }
 
+                    launch {
+                        val videoCapture = useCaseGroup.getVideoCapture()
+                        var recording: Recording? = null
+                        for (event in videoCaptureControlEvents) {
+                            when (event) {
+                                is VideoCaptureControlEvent.StartRecordingEvent -> {
+                                    if (videoCapture == null) {
+                                        throw RuntimeException(
+                                            "Attempted video recording with null videoCapture"
+                                        )
+                                    }
+                                    recording = startVideoRecordingInternal(
```

Comment:
Once a recording is started, we should listen to changes in the `flashMode` and `audioMuted` fields from `transientSettings` to determine whether we should turn on/off the flash and mute/unmute the audio. This can be done by launching another coroutine in this `when` block that listens to `transientSettings`. Creating a suspend version of `PendingRecording.startRecording` that is similar to how `ProcessCameraProvider.runWith` works would make this easier. We can discuss more offline if you're interested.

Reply:
Per offline discussion. Thanks for the example and updated on commit#3.